relay-conventions/build/name.rs

```diff
@@ -47,52 +47,30 @@ pub fn name_file_output(names: impl Iterator<Item = Name>) -> TokenStream {
             let if_clauses = parts.iter().flat_map(|part| {
                 if let TemplatePart::Attribute(name, ident) = part {
                     Some(quote! {
-                        let Some(#ident) = attributes.get_value(#name).and_then(val_to_string)
+                        let Some(#ident) = attributes.get_value(#name)
+                            && matches!(#ident, Val::String(_) | Val::Bool(_) | Val::U64(_) | Val::I64(_) | Val::F64(_))
                     })
                 } else {
                     None
                 }
             });
 
-            // Then, we allocate a string with the appropriate size to hold the name.
-            let declare_name_string = {
-                let literal_length = parts.iter().fold(0, |acc, part|{
-                    if let TemplatePart::Literal(s) = part {
-                        acc + s.len()
-                    } else {
-                        acc
-                    }
-                });
-                let attribute_lengths = parts.iter().flat_map(|part|
-                    if let TemplatePart::Attribute(_, ident) = part {
-                        Some(quote! { #ident.len() })
-                    } else {
-                        None
-                    });
-                quote! {
-                    #[allow(clippy::identity_op)]
-                    let mut name = String::with_capacity(#literal_length + #(#attribute_lengths)+*);
-                }
-            };
-
-            // Finally, append each template part (literal or dynamic) to the string.
-            let append_parts_to_name = parts.iter().map(|part| match part {
-                TemplatePart::Literal(s) => {
-                    quote! {
-                        #[allow(clippy::single_char_add_str)]
-                        name.push_str(#s);
-                    }
+            // Then, construct the format string and argument list for a `format!` call to produce the name.
+            let format_string = parts.iter().map(|part| match part {
+                TemplatePart::Literal(l) => *l,
+                TemplatePart::Attribute(_, _) => "{}",
+            }).collect::<Vec<&str>>().join("");
```

Comment:
Should be able to just collect into a `String`.

```suggestion
            }).collect::<String>();
```

Reply:
Oh, that's really nice. Thanks!